pubsub/spring/src/main/java/demo/PubSubApplication.java

```diff
@@ -0,0 +1,149 @@
+  @Bean
+  public PubSubInboundChannelAdapter inboundChannelAdapter(
+      @Qualifier("inputMessageChannelForSubOne") MessageChannel messageChannel,
+      PubSubTemplate pubSubTemplate) {
+    PubSubInboundChannelAdapter adapter =
+        new PubSubInboundChannelAdapter(pubSubTemplate, "sub-one");
+    adapter.setOutputChannel(messageChannel);
+    adapter.setAckMode(AckMode.MANUAL);
+    adapter.setPayloadType(String.class);
+    return adapter;
+  }
+
+  // Define what happens to the messages arriving in the message channel.
+  @ServiceActivator(inputChannel = "inputMessageChannelForSubOne")
+  public void messageReceiver(
+      String payload,
+      @Header(GcpPubSubHeaders.ORIGINAL_MESSAGE) BasicAcknowledgeablePubsubMessage message) {
+    LOGGER.info("Message arrived via an inbound channel adapter from sub-one! Payload: " + payload);
+    message.ack();
+  }
+  // [END pubsub_spring_inbound_channel_adapter]
+
+  // [START pubsub_spring_outbound_channel_adapter]
+  // Create an outbound channel adapter to send messages from the input message channel to the
+  // topic `topic-two`.
+  @Bean
+  @ServiceActivator(inputChannel = "inputMessageChannelForSubOne")
+  public MessageHandler messageSender(PubSubTemplate pubsubTemplate) {
+    PubSubMessageHandler adapter = new PubSubMessageHandler(pubsubTemplate, "topic-two");
+
+    adapter.setPublishCallback(
+        new ListenableFutureCallback<String>() {
+          @Override
+          public void onFailure(Throwable throwable) {
+            LOGGER.info("There was an error sending the message.");
+          }
+
+          @Override
+          public void onSuccess(String result) {
+            LOGGER.info("Message was sent via the outbound channel adapter to topic-two!");
+          }
+        });
+    return adapter;
+  }
+  // [END pubsub_spring_outbound_channel_adapter]
+
+  // [START pubsub_spring_cloud_stream_input_binder]
+  // Create an input binder to receive messages from `topic-two` using a Consumer bean.
+  @Bean
+  public Consumer<Message<String>> receiveMessageFromTopicTwo() {
+    return message -> {
+      LOGGER.info(
+          "Message arrived via an input binder from topic-two! Payload: " + message.getPayload());
+    };
+  }
+  // [END pubsub_spring_cloud_stream_input_binder]
+
+  // [START pubsub_spring_cloud_stream_output_binder]
+  // Create an output binder to send messages to `topic-one` using a Supplier bean.
+  @Bean
+  public Supplier<Flux<Message<String>>> sendMessageToTopicOne() {
+    return () ->
+        Flux.fromStream(
+                // Generate a stream that sends a numbered message every 10 seconds.
+                Stream.generate(
+                    new Supplier<Message<String>>() {
+                      @Override
+                      public Message<String> get() {
+                        try {
+                          Thread.sleep(10000);
+                        } finally {
+                          Random rand = new Random();
```

Comment:
Ok, but shouldn't this be somewhere earlier - possibly as a classvar or static so it gets reused?

Reply:
Sounds good. Updated.